tools/profile-endpoints.py

```diff
@@ -0,0 +1,330 @@
+import subprocess
+import requests
+import time
+
+# Constants
+UUID = "c672c3c9-ab54-4e19-827c-f2af329da814"
+DATAREPO_URL = "https://jade.datarepo-dev.broadinstitute.org"
+
+
+def run_command(command):
+    """
+    Runs a shell command and captures the output.
+
+    Args:
+        command (str): The shell command to run.
+
+    Returns:
+        str: The command output if successful; otherwise, None.
+    """
+    try:
+        result = subprocess.run(command, shell=True, capture_output=True, text=True)
+        if result.returncode == 0:
+            return result.stdout.strip()
+        else:
+            raise Exception(
+                f"Command failed with exit code {result.returncode}: {result.stderr}"
+            )
+    except Exception as e:
+        print(f"Error executing command: {e}")
+        return None
+
+
+def authenticate():
+    """
+    Prompts the user to authenticate using gcloud auth login.
+    """
+    print("Please complete the authentication in the web browser that opens.")
+    auth_command = "gcloud auth login"
+    run_command(auth_command)
+
+
+def get_access_token():
+    """
+    Retrieves the access token using gcloud auth print-access-token.
+
+    Returns:
+        str: The access token if successful; otherwise, None.
+    """
+    token_command = "gcloud auth print-access-token"
+    return run_command(token_command)
+
+
+# HTTP Request Functions
+
+
+def make_get_request(endpoint_url, token):
```

Comment:
this should be `get_time_taken_for_get_request` or something like that, I'm going to do some refactoring, I wanted to get this to a working place before I did some cleaning!